seccomp/umount.go

```diff
@@ -117,15 +124,81 @@ func (u *umountSyscallInfo) process() (*sysResponse, error) {
 	return u.tracer.createContinueResponse(u.reqId), nil
 }
 
+// processUmmountAllowed purpose is to prevent immutable resources from being
+// unmounted.
+//
+// Method will return 'true' if the umount operation is deemed legit, and will
+// return 'false' otherwise.
+func (u *umountSyscallInfo) processUmountAllowed(
+	mip domain.MountInfoParserIface) (bool, *sysResponse) {
+
+	// There must be mountinfo state present for this target. Otherwise, let
+	// kernel handle the error.
+	info := mip.GetInfo(u.Target)
```

Comment:
The `u.Target` is the target (absolute) path with respect to the process' root (which may not be necessarily the sys container's root). 

It seems to me we have to convert that to the target with respect to the sys container's root.  The reason is that the mountInfoParser has the paths with respect to the sys container's root (correct?). 

If we don't adjust `u.Target` accordingly, then the code will get confused. For example, a process that chroots to `/var/lib` and then unmounts `/docker` will be allowed to unmount. That's because `u.Target = /docker`, so `mip.GetInfo()` will return `nil`.

If we do adjust `u.Target` as I am suggesting, then it seems to me that the `if p.RootInode() != u.cntr.InitProc().RootInode()` check below is no longer needed.

Reply:
The problem is that we don't have a way of adjusting u.Target relative to the sys container's root, because the kernel does not provide the info that will allow us to do this.

So what we need to do instead, it seems to me, is to have the mountInfoParser store the inode of each immutable mount. This way, when u.Target comes in, we can get it's inode (via `stat()`), and compare it unequivocally with the inodes for the immutable mounts. This works regardless of whether the process doing the unmount chroote'd or not. 
It also allows us to not care about whether the process unshared or not it's namespace.

Reply:
Ok got it, my mistake.